src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -1423,9 +1437,9 @@ private void SendRenewFault(RequestContext requestContext, SecurityProtocolCorre
                     {
                         PrepareReply(message, response);
                         TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
-                        response = _securityProtocol.SecureOutgoingMessage(response, timeoutHelper.GetCancellationToken());
+                        response = SecurityProtocol.SecureOutgoingMessage(response, timeoutHelper.GetCancellationToken());
                         response.Properties.AllowOutputBatching = false;
-                        SendMessage(requestContext, response, timeoutHelper.GetCancellationToken());
+                        _ = SendMessageAsync(requestContext, response, timeoutHelper.GetCancellationToken());
                     }
```

Comment:
SendMessageAsync will return an uncompleted Task and then the finally code will execute closing the response potentially before it's finished being used. This also prevents exceptions from being bubbled up.

Reply:
As discussed in meeting, for failure code paths such as this, it's okay to use GetAwaiter.GetResult() if switching the code path to all async would be overly disruptive or too big a change. This only applies for error or failure code paths. The happy case code paths should always be made async. Please add a short comment explaining using GetResult as this is an error code path and we might revisit to make async at a later time.